common/src/main/java/org/apache/falcon/security/DefaultAuthorizationProvider.java

```diff
@@ -294,15 +296,16 @@ protected boolean isUserInAdminGroups(UserGroupInformation proxyUgi) {
 
     protected void authorizeEntityResource(UserGroupInformation authenticatedUGI,
                                            String entityName, String entityType,
-                                           String action)
+                                           String action, boolean entityTypeOptional)
         throws AuthorizationException, EntityNotRegisteredException {
-
-        Validate.notEmpty(entityType, "Entity type cannot be empty or null");
+        if (!entityTypeOptional) {
+            Validate.notEmpty(entityType, "Entity type cannot be empty or null");
+        }
         LOG.debug("Authorizing authenticatedUser={} against entity/instance action={}, "
                 + "entity name={}, entity type={}",
                 authenticatedUGI.getShortUserName(), action, entityName, entityType);
 
-        if (entityName != null) { // lifecycle actions
+        if (entityType != null && entityName != null) { // lifecycle actions
```

Comment:
@bvellanki We already have logs if this doesn't hold:
{code}
        if (entityType != null && entityName != null) { // lifecycle actions
.......
        } else {
            // non lifecycle actions, lifecycle actions with null entity will validate later
            LOG.info("Authorization for action={} will be done in the API", action);
        }
{code}
We also log the action, entity name, and entity type before this IF check. So by looking at the present log, we would be able to know why we skip authorization for an action that doesn't have type and name information.